server/middleware/auth.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-else-return */
+/* eslint-disable consistent-return */
+/* eslint-disable no-console */
+import jwt from 'jsonwebtoken';
+import config from '../config';
+
+function loginResponse(response) {
+  return response.status(401).json({
+    success: false,
+    message: 'Please login first',
+  });
+}
+
+function unauthorizedResponse(response) {
+  return response.status(401).json({
+    success: false,
+    message: 'unauthorized',
+  });
+}
+
+// make sure user is logged in - Authentication
+exports.loginRequired = function (request, response, next) {
+  try {
+    console.log(request.headers.authorization);
+    const token = request.headers.authorization.split(' ')[1]; // Bearer <token>
```

Comment:
write this code 
`const token = request.headers.authorization && request.headers.authorization.split(' ')[1];
if(!token){
    return res.status(400).json({ success: false, message: 'missing token header'});
}
`

Reply:
@avinash360 Why do we need to add `'Bearer '` before the token? We are splitting and using only the token anyway.
Looks like we are doing unnecessary work at both server and client.

Reply:
@Navneet-Sahota  Read this:
https://stackoverflow.com/a/9879273